tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/QueryFixerMain.java

```diff
@@ -1,19 +1,196 @@
 
     if (args.length == 0) {
-      // TODO: provide a more actionable info. This will be done once we finalize the input parameters.
-      logger.atInfo().log("not enough arguments");
+      System.out.println("Please provide arguments.");
+      printHelpAndExit(options);
+    }
+
+    CommandLineParser parser = new DefaultParser();
+    try {
+      return parser.parse(options, args);
+    } catch (ParseException e) {
+      System.out.println(e.getMessage());
+      printHelpAndExit(options);
+      return null;
+    }
+  }
+
+  public static void printHelpAndExit(Options options) {
+    HelpFormatter formatter = new HelpFormatter();
+    formatter.printHelp("-opt <value> --long-opt <value> \"query\"", options);
+    System.exit(1);
+  }
+
+  public static void main(String[] args) {
+    CommandLine cmd = readFlags(args);
+
+    String credentialPath = cmd.getOptionValue("credential");
+    String projectId = cmd.getOptionValue("project-id");
+    BigQueryOptions bigQueryOptions = buildBigQueryOptions(credentialPath, projectId);
+
+    if (cmd.getArgList().isEmpty()) {
+      // In CLI mode, all the instructions are output by print functions, because logger outputs
+      // extract info (time, code position) that distracts users.
+      System.out.println(
+          "Please provide the query as an argument, enclosed by double quote. Use --help for instruction.");
+    }
+
+    String query = cmd.getArgList().get(0);
+    System.out.println("Input query: " + query);
+
+    AutomaticQueryFixer queryFixer = new AutomaticQueryFixer(bigQueryOptions);
+
+    String interactMode = cmd.getOptionValue("interact");
+    if (interactMode == null || interactMode.equalsIgnoreCase("none")) {
+      // todo: Implement Non-interactive mode
+      FixResult fixResult = queryFixer.fix(query);
+      if (fixResult.getOptions().isEmpty()) {
+        return;
+      }
+      String newQuery = fixResult.getOptions().get(0).getFixedQuery();
+      printQueryResult(newQuery, bigQueryOptions);
+
+    } else if (interactMode.equalsIgnoreCase("guide")) {
+      // todo: Implement guide mode
+      return;
+    } else if (interactMode.equalsIgnoreCase("all") || interactMode.equalsIgnoreCase("full")) {
+      FixResult fixResult = fullInteractMode(queryFixer, query);
+      printFixResult(fixResult, cmd.getOptionValue("output"));
+    } else {
+      System.out.println("Interact Mode (-i) is incorrect. Use --help for usage.");
+      System.exit(1);
+    }
+  }
+
+  private static BigQueryOptions buildBigQueryOptions(String credentialPath, String projectId) {
+    if (credentialPath == null) {
+      return BigQueryOptions.newBuilder().setProjectId(projectId).build();
+    } else {
+      // TODO: should support this in near future.
+      System.out.println("customized credential is not supported");
+      System.exit(1);
+      return null;
+    }
+  }
+
+  private static FixResult fullInteractMode(AutomaticQueryFixer queryFixer, String query) {
```

Comment:
nit: "fullInteractMode" -> "fixQueryInFullInteractMode"? maybe "fixQueryInInteractiveMode"?

Reply:
change to `fixQueryInFullInteractMode`.